cmake/kokkos_tribits.cmake

```diff
@@ -160,6 +160,11 @@ FUNCTION(KOKKOS_ADD_EXECUTABLE_AND_TEST ROOT_NAME)
             )
         ENDIF()
     ENDIF()
+    # We noticed problems with -fvisibility-hidden for inline static variables
+    # if Kokkos was built as shared library.
+    IF(BUILD_SHARED_LIBS AND NOT KOKKOS_CXX_HOST_COMPILER_ID MATCHES MSVC)
+      TARGET_COMPILE_OPTIONS(${PACKAGE_NAME}_${ROOT_NAME} PRIVATE -fvisibility=hidden)
+    ENDIF()
```

Comment:
Why did you decide to do this instead of using `CMAKE_CXX_VISIBILITY_PRESET=hidden` and `CMAKE_VISIBILITY_INLINES_HIDDEN=ON`?

Reply:
I wasn't aware of `CMake` attributes using them now.